opentelemetry-sdk/src/metrics/view.rs

```diff
@@ -102,19 +102,25 @@ impl View for Box<dyn View> {
 /// ```
 pub fn new_view(criteria: Instrument, mask: Stream) -> Result<Box<dyn View>> {
     if criteria.is_empty() {
-        global::handle_error(MetricsError::Config(format!(
-            "no criteria provided, dropping view. mask: {mask:?}"
-        )));
+        otel_warn!(
```

Comment:
for a method that returns `Result`, I don't think there is any need of internal logging. If we really want to do some log, then it should be Debug level.

Reply:
The actual error is getting lost in all these cases, as we just return success with empty view.

Reply:
if we always return success, then that looks like a flaw. Since it is not introduced in this PR, lets remove the error logging from here, and put a todo (optional). We need to rework the View and can fix these then.

Reply:
Added TODO, and removed logging.